token.go

```diff
@@ -0,0 +1,22 @@
+package auth
+
+import (
+	"github.com/go-martini/martini"
+	"net/http"
+)
+
+// AuthToken is the authorization token extracted from the request.
+type AuthToken string
+
+// TokenFunc returns a Handler that authenticates via an X-Token header using the provided function.
+// The function should return true for a valid token.
+func TokenFunc(authfn func(string) bool) martini.Handler {
+	return func(res http.ResponseWriter, req *http.Request, c martini.Context) {
+		auth := req.Header.Get("X-Token")
```

Comment:
Nah, I don't really care that much. Either way is fine with me.

If @codegangsta if fine with this patch, I'd just add a few simple tests before we merge, that's all.